data/abilities.js

```diff
@@ -1496,7 +1489,8 @@ let BattleAbilities = {
 		shortDesc: "On switch-in, this Pokemon Transforms into the opposing Pokemon that is facing it.",
 		onStart(pokemon) {
 			if (this.activeMove && this.activeMove.id === 'skillswap') return;
-			let target = pokemon.side.foe.active[pokemon.side.foe.active.length - 1 - pokemon.position];
+			const possibleTargets = pokemon.foes(false, true);
```

Comment:
I often wonder if these should be different functions, rather than the same function with different booleans passed to them. How much code do they even share?

Reply:
The first boolean says "filter out non adjacent foes". The second boolean says "don't filter out `null` or fainted foes. All the code that actually goes into finding what we're filtering is the same.

Reply:
The only thing I don't like about this being two booleans is that there's no use case for `pokemon.foes(true, true)`

Reply:
I can't think of a more intuitive alternative besides using `string` codes, though, and I don't like codes for this since it wouldn't fit nicely into a switch statement. After all, it's not "do a, b, or c" it's "do a, b, or a and b".